source/beamline/CLS/CLSBeamlineStatus.h

```diff
@@ -42,6 +42,8 @@ class CLSBeamlineStatus : public CLSBiStateGroup
 	/// Returns the beam on value for the given control.
 	double componentBeamOnValue(AMControl *control) const { return controlState1ValueMap_.value(control, -1); }
 
+	/// Returns the beamline status PV
+	AMControl * beamlineStatusPVControl() { return beamlineStatusPVControl_ ? beamlineStatusPVControl_ : this; }
```

Comment:
I think this should return the control (which is initialized to 0) and returns 0 if not set.  Then the things that use it just have a piece of logic that is:

```c++
if (control->beamlineStatusPVControl())
    control->beamlineStatusPVControl()->stuff();
else
    control->stuff();
```